### PR TITLE
Timeseries: add UI to control the span nulls threshold

### PR DESCRIPTION
**Pull request by ryantxu**
Extracted from #32146

Values are always stored as milliseconds, but displayed in formatted time intervals

**After:**
image

image

**Before:**
image

### CODE REVIEW

File: public/app/plugins/panel/timeseries/SpanNullsEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  {
+    label: 'Threshold',
+    value: 3600000, // 1h
+  },
+];
+
+export const SpanNullsEditor: React.FC<FieldOverrideEditorProps<boolean | number, any>> = ({ value, onChange }) => {
+  const isThreshold = typeof value === 'number';
+  const formattedTime = isThreshold ? secondsToHms((value as number) / 1000) : undefined;
+  GAPS_OPTIONS[2].value = isThreshold ? (value as number) : 3600000; // reuse constants
```
Comment (leeoniya) [inline public/app/plugins/panel/timeseries/SpanNullsEditor.tsx:24]: can probably pull this const up :)

**Comment by leeoniya** [comment]
works as expected :+1:

**Review by leeoniya — APPROVED** [review]

**Comment by achatterjee-grafana** [comment]
Hi @ryantxu, would there any additional documentation impact for this change? For example, should we capture new/updated images, etc.? Also, is this change for the v8.0 or v7.5 release? 
Thank you.

**Comment by oddlittlebird** [comment]
@ryantxu, this documentation does not match the Granana 7.5 UI. Also, you removed some periods from the end of sentences. Please revert these doc changes. If they are for Grafana 8, then they are way too early.

**Comment by ryantxu** [comment]
dooh -- I thought i was being a good citizen, but forgot that master docs != master code.

This is for v8 unless the noisy customer really needs it for 7.5.1...

I'll revert, and I guess we will make sure everything is in sync for v8

**Comment by achatterjee-grafana** [comment]
Can we also have the version label added to the PR. Thanks!

**Comment by oddlittlebird** [comment]
Except for not waiting for documentarian review, you are doing the right thing technically @ryantxu. It's just that the system cannot handle it until the issue to not have the master branch docs pushed every time gets addressed.

**Comment by oddlittlebird** [comment]
@achatterjee-grafana, maybe we should start a Gdoc or a draft to capture these 8.0 updates. Let's talk tomorrow.

**Comment by osg-grafana** [comment]
> @achatterjee-grafana, maybe we should start a Gdoc or a draft to capture these 8.0 updates. Let's talk tomorrow.

When can we start putting 8.0 documentation into source control?

**Pull request merged**